src/hooks/useInitialValues.test.ts

```diff
@@ -333,4 +334,239 @@ describe('useInitialValues', () => {
+    let hook = null;
+    const fieldWithCalculateExpression: FormField = {
+      label: 'Latest mother HIV status',
+      type: 'obs',
+      questionOptions: {
+        rendering: 'fixed-value',
+        concept: 'af7c1fe6-d669-414e-b066-e9733f0de7a8',
+        calculate: {
+          calculateExpression:
+            "resolve(api.getLatestObs(patient.id, '159427AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', '2549af50-75c8-4aeb-87ca-4bb2cef6c69a'))?.valueCodeableConcept?.coding[0]?.code",
+        },
+      },
+      id: 'latest_mother_hiv_status',
+    };
+    allFormFields.push(fieldWithCalculateExpression);
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...allFormFields],
+          null,
+          false,
+          {
+            encounter: undefined,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+    expect(initialValues['latest_mother_hiv_status']).toBe('664AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+  });
+
+  it('should fall back to encounter value if the calculated expression result is null or undefined', async () => {
+    let hook = null;
+    let formFields: Array<FormField> = [
+      {
+        label: 'Height (cm)',
+        type: 'obs',
+        required: false,
+        id: 'height',
+        questionOptions: {
+          rendering: 'number',
+          concept: '5090AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          answers: [],
+        },
+      },
+      {
+        label: 'Weight (Kgs)',
+        type: 'obs',
+        required: false,
+        id: 'weight',
+        questionOptions: {
+          rendering: 'number',
+          concept: '5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          answers: [],
+        },
+        validators: [],
+      },
+      {
+        label: 'BMI:Kg/M2 (Function calcBMI | useFieldValue)',
+        type: 'obs',
+        required: false,
+        id: 'bmi',
+        questionOptions: {
+          rendering: 'number',
+          defaultValue: 0,
+          concept: '1342AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          isTransient: true,
+          disallowDecimals: false,
+          calculate: {
+            calculateExpression: 'calcBMI(heigh,weight)',
+          },
+        },
+        validators: [],
+        questionInfo: 'this calculates BMI using calcBMI function and useFieldValue of weight and height',
+      },
+    ];
+
+    await act(async () => {
+      hook = renderHook(() =>
```

Comment:
We have this block identical or nearly identical many times in this file. Could you please refactor it out into a helper function?